types/react-select/src/styles.d.ts

```diff
@@ -1,63 +1,35 @@
-import {
-  containerCSS,
-  indicatorsContainerCSS,
-  valueContainerCSS,
-} from './components/containers';
-import { css as controlCSS } from './components/Control';
-import { groupCSS, groupHeadingCSS } from './components/Group';
-import {
-  clearIndicatorCSS,
-  dropdownIndicatorCSS,
-  loadingIndicatorCSS,
-  indicatorSeparatorCSS,
-} from './components/indicators';
-import { inputCSS } from './components/Input';
-import { placeholderCSS } from './components/Placeholder';
-import { optionCSS } from './components/Option';
-import {
-  menuCSS,
-  menuListCSS,
-  menuPortalCSS,
-  noOptionsMessageCSS,
-  loadingMessageCSS,
-} from './components/Menu';
-import { css as singleValueCSS } from './components/SingleValue';
-import {
-  multiValueCSS,
-  multiValueLabelCSS,
-  multiValueRemoveCSS,
-} from './components/MultiValue';
 import { CSSProperties } from 'react';
+import { ControlProps } from './components/Control';
 
 export interface Props { [key: string]: any; }
 
-export type styleFn = (base: CSSProperties, state: any) => CSSProperties;
+export type StylesConfigFunction<Props = any> = (base: CSSProperties, props: Props) => CSSProperties;
```

Comment:
Maybe consider leaving `styleFn` in and `/*@deprecated*/` it to ease the transition for anyone who might be importing it now, but I leave it up to you.